### PR TITLE
Cleaning up default config formatting, values and comments

### PR DESCRIPTION
**Pull request by drewoldag**
Doing my best to apply some uniform formatting to the config file to make it a little less overwhelming. 

The linting rules that I tried to adhere to:
1. No table-level comments
2. No comments outside of a table
3. Two lines between tables
4. One line between a key and the following comment
5. 2 lines of comments "maximum" per key - I failed on this for the test/validation/train splits.
6. No free floating blocks of comments - Again I failed when it came to the test/validation/train splits, there's just a lot there.
7. Reasonable maximum line length for comments
8. Include optional values when possible (i.e. allowed log levels)
9. Tried to have one comment per key, but used best judgement there. (i.e. username, password only got one comment)

### CODE REVIEW

File: src/fibad/fibad_default_config.toml [hunk 1]
```diff
@@ -1,172 +1,189 @@
...
-# but can be configured to a common location for batch processing.
-credentials_file = "./credentials.ini"
 
-num_sources = -1 # Values below 1 here indicate all sources in the catalog will be downloaded
+# The number of sources to download from the catalog. Default is -1, which
+# downloads all sources in the catalog.
+num_sources = -1
+
+# The number of sources to skip in the catalog. Default is 0.
 offset = 0
```
Comment (mtauraso) [inline src/fibad/fibad_default_config.toml:51]: This shouldn't be a config anymore

File: src/fibad/fibad_default_config.toml [hunk 2]
```diff
@@ -1,172 +1,189 @@
...
 [optimizer]
-# The name of the built-in optimizer to use or the libpath to an external optimizer
+# The name of the built-in optimizer to use or the import path to an external optimizer
 name = "torch.optim.SGD"
 
-# Default PyTorch optimizer parameters. The keys match the names of the parameters
+# learning rate parameter for default optimizer. The keys match the names of the parameters
 lr = 0.01
+
+# momentum parameter for default optimizer. The keys match the names of the parameters
```
Comment (mtauraso) [inline src/fibad/fibad_default_config.toml:107]: "The keys match the names of the parameters" made sense before. It doesn't now.

I think we want to include the prior meaning.

File: src/fibad/fibad_default_config.toml [hunk 3]
```diff
@@ -1,172 +1,189 @@
...
 crop_to = false
 
-# Limit data loader to only particular filters when there are more in the data set.
-#
-# When not provided by the user, the number of filters will be automatically gleaned from the data set.
-# Defaults behavior is produced by the false value.
-#
-#filters = ["HSC-G", "HSC-R", "HSC-I", "HSC-Z", "HSC-Y"]
+# Specific to `hsc_data_set`. Limit to only particular filters from the data set.
+# When `false`, uses all filters found in the data set.
```
Comment (mtauraso) [inline src/fibad/fibad_default_config.toml:136]: I think the commented out example was helpful here. I'd advocate for its return.

File: src/fibad/fibad_default_config.toml [hunk 4]
```diff
@@ -1,172 +1,189 @@
...
+# [general].data_dir. Implementation is data_set class dependent. Use `false` for no filtering.
 filter_catalog = false
 
-# How to split the data between training and eval sets.
-# The semantics are borrowed from scikit-learn's train-test-split, and HF Dataset's train-test-split function
-# It is an error for these values to add to more than 1.0 as ratios or the size of the dataset if expressed
-# as integers.
+# For train_size, validation_size, and test_size, the following options are available:
+# A `float` between `0.0` and `1.0` is the proportion of the dataset to include in the split.
+# If `int`, represents the absolute number of samples in the particular split.
```
Comment (mtauraso) [inline src/fibad/fibad_default_config.toml:145]: This should be "An `int`..." to preserve parallel construction with the line above about 'float' typed values.

File: src/fibad/fibad_default_config.toml [hunk 5]
```diff
@@ -1,172 +1,189 @@
...
+# It is an error for these values to add to more than 1.0 as ratios or the size
+# of the dataset if expressed as integers.
```
Comment (mtauraso) [inline src/fibad/fibad_default_config.toml:147]: My preference would be to rewrite this preamble as follows:

```
# train_size, validation_size, and test_size use these conventions:
# * A `float` between `0.0` and `1.0` is the proportion of the dataset to include in the split.
# * An `int` representing the absolute number of samples in the particular split.
# * It is an error for these values to add to more than 1.0 as ratios or the size
#   of the dataset if expressed as integers.
```

File: src/fibad/fibad_default_config.toml [hunk 6]
```diff
@@ -1,172 +1,189 @@
...
-# Number to seed with for generating a random split. False means the data will be seeded from
-# a system source at runtime.
+# Number to seed with for generating a random split. Use `false` to seed from a
+# system source at runtime.
 seed = false
 
-#Controls whether images are cached during data loading. For training, this reduces runtimes
-#after the first epoch.
+# If `true`, cache data in memory during training. For training, this reduces runtimes
+#after the first epoch. Set to `false` to disable caching or when running inference.
```
Comment (mtauraso) [inline src/fibad/fibad_default_config.toml:169]: This should probably have a note that most users want this off, because it only speeds up inference if you're doing something like drop-out.

File: src/fibad/fibad_default_config.toml (hunk 2 above)
Comment (drewoldag) [reply src/fibad/fibad_default_config.toml:107]: I have an other PR implementing the subtables for criterion and optimizer that should address this. Going to leave it as is for now.

File: src/fibad/fibad_default_config.toml [hunk 7]
```diff
@@ -1,172 +1,189 @@
...
 use_cache = true
 
+
 [data_loader]
-# Default PyTorch DataLoader parameters
-batch_size =32
+# The number of data points to load at once.
+batch_size = 32
```
Comment (mtauraso) [inline src/fibad/fibad_default_config.toml:175]: I'm of the opinion that this should be (128-1024) to avoid the trap of low gpu performance out of the box.

File: src/fibad/fibad_default_config.toml [hunk 8]
```diff
@@ -1,172 +1,189 @@
...
 validate_size = 0.2
 
-# test_size: Size of the test split
-# If `float`, should be between `0.0` and `1.0` and represent the proportion of the dataset to include in the 
-# test split.
-# If `int`, represents the absolute number of test samples.
-# If `false`, the value is set to the complement of the train size.
-# If `train_size` is also `false`, it will be set to `0.25`.
+# Size of the test split.
+# If `false`, the value is set to the complement of train_size.
```
Comment (mtauraso) [inline src/fibad/fibad_default_config.toml:160]: I realize this error existed in the original text, but this is not exactly what we do anymore. I think the current behavior is more like:

```
# if `false`, the value is set to the complement of train_size plus validate_size.
# If `false` and `train_size = false`, test_size is set to `0.25`.
```

File: src/fibad/fibad_default_config.toml [hunk 9]
```diff
@@ -1,172 +1,189 @@
...
+# It is an error for these values to add to more than 1.0 as ratios or the size
+# of the dataset if expressed as integers.
 
-# train_size: Size of the train split
-# If `float`, should be between `0.0` and `1.0` and represent the proportion of the dataset to include in the 
-# train split.
-# If `int`, represents the absolute number of train samples.
-# If `false`, the value is automatically set to the complement of the test size.
+# Size of the train split.
+# If `false`, the value is automatically set to the complement of test_size.
```
Comment (mtauraso) [inline src/fibad/fibad_default_config.toml:150]: This error existed in the original text, but this is not exactly what we do anymore. The current behavior is more like:
```
# If `false`, the value is automatically set to the complement of test_size plus the validate_size (if any).
```

**Comment by mtauraso** [comment]
Left several comments. I think we need to make sure these are fully accurate before merging.

This is also an opportunity to remove configs that serve no purpose and clutter the file. I believe `offset` needs removal, and I suspect `shuffle` does as well.

File: src/fibad/fibad_default_config.toml [hunk 10]
```diff
@@ -1,172 +1,189 @@
...
 [data_loader]
-# Default PyTorch DataLoader parameters
-batch_size =32
+# The number of data points to load at once.
+batch_size = 32
 
-# We could remove this potentially - pytorch-ignite will default to shuffle=True
-# If the user wanted to explicitly require no shuffling, they could set this to false.
+# Whether the input data is shuffled before being loaded.
 shuffle = false
```
[the remark this answers is not in the document]
Comment (drewoldag) [reply src/fibad/fibad_default_config.toml:178]: I'm interested in keeping it for reproducibility purposes. We could probably set the value to `true` to give the expected behavior out of the box. And then it will provide the place for someone to specify `false` if that's what they need.

[the remark this answers is not in the document]
Comment (drewoldag) [reply src/fibad/fibad_default_config.toml:178]: Actually, never mind. Using samplers in the data sets prevents us from using shuffle, so I'll remove this.

File: src/fibad/fibad_default_config.toml (hunk 7 above)
Comment (drewoldag) [reply src/fibad/fibad_default_config.toml:175]: Bumped to 512

File: src/fibad/fibad_default_config.toml (hunk 8 above)
Comment (drewoldag) [reply src/fibad/fibad_default_config.toml:160]: Updated

File: src/fibad/fibad_default_config.toml (hunk 9 above)
Comment (drewoldag) [reply src/fibad/fibad_default_config.toml:150]: Updated

File: src/fibad/fibad_default_config.toml (hunk 5 above)
Comment (drewoldag) [reply src/fibad/fibad_default_config.toml:147]: This looks fine enough for now. We can go into more detail in RTD documentation.

File: src/fibad/fibad_default_config.toml (hunk 4 above)
Comment (drewoldag) [reply src/fibad/fibad_default_config.toml:145]: Done

File: src/fibad/fibad_default_config.toml [hunk 11]
```diff
@@ -1,172 +1,189 @@
...
-# Implementation is dataset class dependent. Default is false meaning now filtering.
+# Path to a fits file that specifies object IDs to use from the data stored in
+# [general].data_dir. Implementation is data_set class dependent. Use `false` for no filtering.
 filter_catalog = false
 
-# How to split the data between training and eval sets.
-# The semantics are borrowed from scikit-learn's train-test-split, and HF Dataset's train-test-split function
-# It is an error for these values to add to more than 1.0 as ratios or the size of the dataset if expressed
-# as integers.
+# For train_size, validation_size, and test_size, the following options are available:
```
[the remark this answers is not in the document]
Comment (drewoldag) [reply src/fibad/fibad_default_config.toml:143]: done

File: src/fibad/fibad_default_config.toml (hunk 3 above)
Comment (drewoldag) [reply src/fibad/fibad_default_config.toml:136]: Added the example back

File: src/fibad/fibad_default_config.toml (hunk 1 above)
Comment (drewoldag) [reply src/fibad/fibad_default_config.toml:51]: Removed

File: src/fibad/fibad_default_config.toml (hunk 6 above)
Comment (drewoldag) [reply src/fibad/fibad_default_config.toml:169]: Updated.

**Comment by drewoldag** [comment]
I believe that all the comments have been resolved or addressed at this point. If there aren't other specific comments, I would say that we merge this and fix the one-offs as we find them.
